source/base/tensor_product_polynomials.cc

```diff
@@ -700,6 +704,16 @@ AnisotropicPolynomials<dim>::get_n_tensor_pols(
 }
 
 
+template <int dim>
+std::unique_ptr<ScalarPolynomialsBase<dim>>
+AnisotropicPolynomials<dim>::clone() const
+{
+  Assert(false, ExcNotImplemented());
+
+  return std_cxx14::make_unique<AnisotropicPolynomials<dim>>(*this);
```

Comment:
why is it not implemented? Does the implementation in the last line not work?

Reply:
Great question. I forgot to remove that assert while addressing your previous comments!